packages/quantic/cypress/integration/facets/category-facet/catgory-facet.cypress.ts

```diff
@@ -0,0 +1,200 @@
+import {configure} from '../../../page-objects/configurator';
+
+import {CategoryFacetExpectations as Expect} from './category-facet-expectations';
+import {
+  canadaHierarchy,
+  canadaHierarchyIndex,
+  CategoryFacetActions as Actions,
+} from './category-facet-actions';
+import {InterceptAliases, interceptSearch} from '../../../page-objects/search';
+
+interface CategoryFacetOptions {
+  field: string;
+  label: string;
+  basePath: string;
+  noFilterByBasePath: boolean;
+  noFilterFacetCount: boolean;
+  delimitingCharacter: string;
+  numberOfValues: number;
+  sortCriteria: string;
+  withSearch: boolean;
+  isCollapsed: boolean;
+}
+
+describe('quantic-category-facet', () => {
+  const pageUrl = 's/quantic-category-facet';
+
+  const defaultField = 'geographicalhierarchy';
+  const defaultLabel = 'Country';
+  const defaultNumberOfValues = 8;
+
+  function visitCategoryFacetPage(
+    options: Partial<CategoryFacetOptions> = {},
+    waitForSearch = true
+  ) {
+    interceptSearch();
+    cy.visit(pageUrl);
+    configure(options);
+    if (waitForSearch) {
+      cy.wait(InterceptAliases.Search);
+    }
+  }
+  function setupWithDefaultSettings() {
+    visitCategoryFacetPage(
+      {
+        field: defaultField,
+        label: defaultLabel,
+        numberOfValues: defaultNumberOfValues,
+      },
+      false
+    );
+  }
+
+  function setupGoDeeperOneLevel() {
+    setupWithDefaultSettings();
+    Actions.selectChildValueAt(canadaHierarchyIndex[0]);
```

Comment:
Since `setupGoDeeperOneLevel` doesn't wait for the search query to complete, I would suggest to add this line before selecting the child value.

    cy.wait(InterceptAliases.Search);